co-log-core/src/Colog/Core/Action.hs

```diff
@@ -144,13 +150,68 @@ instance Applicative m => Monoid (LogAction m a) where
 
 #if MIN_VERSION_base(4,12,0)
 instance Contravariant.Contravariant (LogAction m) where
+    contramap :: (a -> b) -> LogAction m b -> LogAction m a
     contramap = cmap
     {-# INLINE contramap #-}
 
+    (>$) :: b -> LogAction m b -> LogAction m a
     (>$) = (Colog.Core.Action.>$)
     {-# INLINE (>$) #-}
 #endif
 
+-- | For tracking usage of unrepresentable class instances of 'LogAction'.
+type family UnrepresentableClass :: Constraint
+  where
+    UnrepresentableClass = TypeError
+        ( 'Text "'LogAction' cannot have a 'Functor' instance by design."
+        ':$$: 'Text "However, the instance is used in here."
```

Comment:
Just a minor rewording, but I'm not sure about the correct words. This is already great :+1: 
```suggestion
        ':$$: 'Text "However, you've attempted to use this instance."
```